modules/apps/site-initializer/site-initializer-raylife-ap/site-initializer-raylife-ap-test/src/testFunctional/tests/RaylifeAPTypeOfInsurance.testcase

```diff
@@ -0,0 +1,133 @@
+		}
+		else {
+			User.logoutPG();
+		}
+	}
+
+	@description = "Verify if the user can swicth between the cards."
+	@priority = 3
+	test IfCanSwitchBetweenCards {
+		property test.name.skip.portal.instance = "RaylifeAPTypeOfInsurance#IfCanSwitchBetweenCards";
+
+		// Skip portal instance due to LRQA-79129. Unskip after it is resolved
+
+		task ("Given the modal for new application is open") {
+			RaylifeAP.goToRaylifeAPSiteWithRefresh();
+
+			Click(
+				key_text = "Application",
+				locator1 = "Button#ANY");
+		}
+
+		task ("Then Assert if the user can switch between the cards") {
+			RaylifeApApplication.modalOption(option = "Business");
+
+			AssertElementPresent(
+				key_text = "Business",
+				locator1 = "RaylifeApplicationModal#MODAL_INSURANCE");
+
+			RaylifeApApplication.modalOption(option = "Personal");
+
+			AssertElementPresent(
+				key_text = "Personal",
+				locator1 = "RaylifeApplicationModal#MODAL_INSURANCE");
+
+			Click(
+				key_text = "Next",
+				locator1 = "Button#ANY");
+
+			RaylifeApApplication.modalOption(option = "Auto");
+
+			AssertElementPresent(
+				key_text = "Auto",
+				locator1 = "RaylifeApplicationModal#MODAL_INSURANCE");
+		}
+	}
+
+	@description = "Verify if when the user selects a card, go to next page and click on 'Previous' button the card still selected (active)."
+	@priority = 4
+	test IfCardStillSelectedWhenUserGoToPreviousPage {
+		property test.name.skip.portal.instance = "RaylifeAPTypeOfInsurance#IfCardStillSelectedWhenUserGoToPreviousPage";
+
+		// Skip portal instance due to LRQA-79129. Unskip after it is resolved
+
+		task ("Given the New Application modal is open") {
+			RaylifeAP.goToRaylifeAPSiteWithRefresh();
+
+			RaylifeApApplication.modalFlow(
+				button = "Previous",
+				insurance = "Personal");
+		}
+
+		task ("Then Assert that the Insurance options is still active after click in 'previous' button") {
+			AssertElementNotPresent(
+				key_text = "Business",
+				locator1 = "RaylifeApplicationModal#MODAL_INSURANCE");
+
+			AssertElementPresent(
+				key_text = "Personal",
+				locator1 = "RaylifeApplicationModal#MODAL_INSURANCE");
+		}
+	}
+
+	@description = "Verify if the 'New Applications' button from the side-bar option 'Applications', works."
+	@priority = 5
+	test IfNewApplicationsFromSideBarWorks {
+		property test.name.skip.portal.instance = "RaylifeAPTypeOfInsurance#IfNewApplicationsFromSideBarWorks";
+
+		// Skip portal instance due to LRQA-79129. Unskip after it is resolved
+
+		task ("Given the modal new application is open by side-bar option") {
+			RaylifeAP.goToRaylifeAPSiteWithRefresh();
+
+			RaylifeAPNavigator.gotoSidebarItem(itemName = "Applications");
+
+			Click(
+				key_text = "New Applications",
+				locator1 = "Button#ANY");
+		}
+
+		task ("Then Assert that the modal is open") {
+			AssertElementPresent(
+				key_text = "What type of insurance does this customer need?",
+				locator1 = "RaylifeApplicationModal#MODAL_DESCRIPTION");
```

Comment:
Hi @styller1, I think that this assert should be a macro. It's good to avoid having functions in the structure of the tests, they must be inside macros. What do you think?